Remotion/Validation/Globalization.UnitTests/ResourceManagerBasedValidationMessageTest.cs

```diff
@@ -15,35 +15,100 @@
+          .Callback(() =>
+          {
+            actualCulture = CultureInfo.CurrentUICulture;
+          })
+          .Returns(true);
+
+      var validationMessage = new ResourceManagerBasedValidationMessage(resourceManagerMock.Object, TestResourceIdentifiers.One);
+      var expectedCulture = CultureInfo.GetCultureInfo("en-US");
+      var formatProvider = CultureInfo.GetCultureInfo("de-AT");
+      var parameters = new object[] { 3123.31};
+
+      var result = validationMessage.Format(expectedCulture, formatProvider, parameters);
+
+      Assert.That(result, Is.EqualTo("€ 3.123,31"));
+      Assert.That(actualCulture, Is.EqualTo(expectedCulture));
     }
 
     [Test]
-    [Ignore("RM-5906")]
+    [SetCulture("it-IT")]
+    [SetUICulture("fr-FR")]
     public void Format_UsesSuppliedCultureInsteadOfCurrentUICulture ()
     {
+      var value = "{0:C}";
+      CultureInfo actualCulture = null;
+
+      var resourceManagerMock = new Mock<IResourceManager>();
+      resourceManagerMock.Setup(_ => _.TryGetString("Remotion.Validation.Globalization.UnitTests.ResourceManagerBasedValidationMessageTest.One", out value))
+          .Callback(() =>
+          {
+            actualCulture = CultureInfo.CurrentUICulture;
+          })
+          .Returns(true);
+
+      var validationMessage = new ResourceManagerBasedValidationMessage(resourceManagerMock.Object, TestResourceIdentifiers.One);
+      var expectedCulture = CultureInfo.GetCultureInfo("en-US");
+      var parameters = new object[] { 3123.31};
+
+      var result = validationMessage.Format(expectedCulture, null, parameters);
+
+      Assert.That(result, Is.EqualTo("¤3,123.31"));
+      Assert.That(actualCulture, Is.EqualTo(expectedCulture));
     }
 
+    [SetCulture("it-IT")]
+    [SetUICulture("fr-FR")]
     [Test]
-    [Ignore("RM-5906")]
-    public void ToString_Override ()
+    public void ToString_ReturnsValueOfResource ()
     {
+      string value = "Remotion.Validation.Globalization.UnitTests.ResourceManagerBasedValidationMessageTest.Two";
+      CultureInfo actualCulture = null;
+
+      var resourceManagerMock = new Mock<IResourceManager>();
+      resourceManagerMock
+          .Setup(_ => _.TryGetString("Remotion.Validation.Globalization.UnitTests.ResourceManagerBasedValidationMessageTest.Two", out value))
+          .Callback(() =>
+          {
+            actualCulture = CultureInfo.CurrentUICulture;
+          })
+          .Returns(true);
+
+      var validationMessage  = new ResourceManagerBasedValidationMessage(resourceManagerMock.Object, TestResourceIdentifiers.Two);
+      var expectedCulture = CultureInfo.GetCultureInfo("fr-FR");
+
+      var result = validationMessage.ToString();
+
+      Assert.That(result, Is.EqualTo("Remotion.Validation.Globalization.UnitTests.ResourceManagerBasedValidationMessageTest.Two"));
```

Comment:
```suggestion
      string value = "Expected Value";
      CultureInfo actualCulture = null;

      var resourceManagerMock = new Mock<IResourceManager>();
      resourceManagerMock
          .Setup(_ => _.TryGetString("Remotion.Validation.Globalization.UnitTests.ResourceManagerBasedValidationMessageTest.Two", out value))
          .Callback(() =>
          {
            actualCulture = CultureInfo.CurrentUICulture;
          })
          .Returns(true);

      var validationMessage  = new ResourceManagerBasedValidationMessage(resourceManagerMock.Object, TestResourceIdentifiers.Two);
      var expectedCulture = CultureInfo.GetCultureInfo("fr-FR");

      var result = validationMessage.ToString();

      Assert.That(result, Is.EqualTo("Expected Value"));
```

It's good practice to use simple strings that are obviously fake values when setting up test doubles. That way, the test is a) shorter and b) there is no chance these values hide incorrect behavior. E.g. you could accidentally return the key instead of the result and then this error would remain undetected.